Firestore/Example/Tests/Local/FSTLevelDBMigrationsTests.mm

```diff
@@ -60,43 +61,54 @@ - (void)tearDown {
 - (void)testAddsTargetGlobal {
   FSTPBTargetGlobal *metadata = [FSTLevelDBQueryCache readTargetMetadataFromDB:_db];
   XCTAssertNil(metadata, @"Not expecting metadata yet, we should have an empty db");
-  [FSTLevelDBMigrations runMigrationsOnDB:_db];
+  LevelDBTransaction transaction(_db, [FSTLevelDB standardReadOptions],
+                                 [FSTLevelDB standardWriteOptions]);
+  [FSTLevelDBMigrations runMigrations:&transaction];
+  transaction.Commit();
   metadata = [FSTLevelDBQueryCache readTargetMetadataFromDB:_db];
   XCTAssertNotNil(metadata, @"Migrations should have added the metadata");
 }
 
 - (void)testSetsVersionNumber {
-  FSTLevelDBSchemaVersion initial = [FSTLevelDBMigrations schemaVersionForDB:_db];
+  LevelDBTransaction transaction(_db, [FSTLevelDB standardReadOptions],
+                                 [FSTLevelDB standardWriteOptions]);
+  FSTLevelDBSchemaVersion initial = [FSTLevelDBMigrations schemaVersion:&transaction];
   XCTAssertEqual(0, initial, "No version should be equivalent to 0");
 
   // Pick an arbitrary high migration number and migrate to it.
-  [FSTLevelDBMigrations runMigrationsOnDB:_db];
-  FSTLevelDBSchemaVersion actual = [FSTLevelDBMigrations schemaVersionForDB:_db];
+  [FSTLevelDBMigrations runMigrations:&transaction];
+  FSTLevelDBSchemaVersion actual = [FSTLevelDBMigrations schemaVersion:&transaction];
   XCTAssertGreaterThan(actual, 0, @"Expected to migrate to a schema version > 0");
 }
 
 - (void)testCountsQueries {
   NSUInteger expected = 50;
-  FSTWriteGroup *group = [FSTWriteGroup groupWithAction:@"Setup"];
-  for (int i = 0; i < expected; i++) {
-    std::string key = [FSTLevelDBTargetKey keyWithTargetID:i];
-    [group setData:"dummy" forKey:key];
+  {
+    LevelDBTransaction transaction(_db, [FSTLevelDB standardReadOptions],
+                                   [FSTLevelDB standardWriteOptions]);
+    for (int i = 0; i < expected; i++) {
+      std::string key = [FSTLevelDBTargetKey keyWithTargetID:i];
+      transaction.Put(key, "dummy");
+    }
+    // Add a dummy entry after the targets to make sure the iteration is correctly bounded.
+    // Use a table that would sort logically right after that table 'target'.
+    std::string dummyKey;
+    // Magic number that indicates a table name follows. Needed to mimic the prefix to the target
+    // table.
+    OrderedCode::WriteSignedNumIncreasing(&dummyKey, 5);
+    OrderedCode::WriteString(&dummyKey, "targetA");
+    transaction.Put(dummyKey, "dummy");
+    transaction.Commit();
+  }
+
+  {
```

Comment:
Rather than putting multiple blocks in a single long test, consider just making separate test methods for this. It will help by forcing you to put a description of what other thing you're testing in the method name :-).

Reply:
There is only one thing being tested here (the final `XCTAssertEqual`), the first block is just setup. The blocks are for automatic cleanup of the transaction and being able to reuse the name.

I'll add a comment to the first block identifying it as setup.